### DISCUSSION TITLE
kubectl: command not found

### DISCUSSION THREAD

**Discussion by imranrazakhan** [k3s-io/k3s#4993] [Q&A]
I have develop a bash script for airgap installation on CentOS 7  but after k3s installation its throwing error command not found for kubectl

```
 #!/bin/bash
 set -o errexit

  cp /k3s /usr/local/bin/
  chmod +x /usr/local/bin/k3s
  chmod a+rx /install.sh

  export INSTALL_K3S_SKIP_DOWNLOAD=true
  export INSTALL_K3S_EXEC=server --cluster-init  --disable servicelb --disable traefik --disable metrics-server --write-kubeconfig ~/.kube/config
  sh /install.sh

  kubectl wait --for=condition=Ready nodes --all --timeout=600s
```

k3s installation is successful but at kubectl command it throw error

**First Try:**

After k3s install added following line, but no success

```
  echo "KUBECONFIG=~/.kube/config:/etc/rancher/k3s/k3s.yaml" >>~/.bashrc
  source ~/.bashrc
```

**Second Try:**
So /usr/local/bin get call again but no success

`source ~/.bash_profile`

**Third Try:**

Change command like below and it work

`/usr/local/bin/kubectl wait --for=condition=Ready nodes --all --timeout=600s`

If i run kubectl directly on shell it works, Please let me know if i am missing something

**Comment by ChristianCiach** [comment]
Just use your third approach. It's the right way!

You are probably running your shellscript as root. On CentOS, `/usr/local/bin` is not on the `PATH` for the root user. 

See https://security.stackexchange.com/questions/136990/whats-the-motivation-for-excluing-usr-local-bin-from-roots-path

**Reply by imranrazakhan** [reply]
@ChristianCiach  I quickly created below script on same server

```
#!/bin/bash
echo $PATH
kubectl wait --for=condition=Ready nodes --all --timeout=600s
```

Output is below

```
# bash test.sh
/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/root/bin
node/xdww condition met
```

kubectl doesn't work only when we call from same script from where we are calling k3s install.sh.

**Reply by brandond** [reply]
You only get a `kubectl` symlink installed in your bin dir if you allow the install script to download k3s for you. If you download the binary directly, you can do one of a couple things:
* run `k3s kubectl`
* manually create a symlink from `/usr/local/bin/kubectl` to `/usr/local/bin/k3s`
* add `/var/lib/rancher/k3s/data/current/bin/` to your $PATH

**Reply by imranrazakhan** [reply]
@brandond Thanks for reply, symlink is already created but its not working when call from same script from where we call k3s install.sh , it works from other scripts/shells, may be k3s install.sh is setting something which is stopping to call or /usr/local/bin cache is not getting update.

```
# cd /usr/local/bin/
# ls -ltrh
total 61M
-rwxr-xr-x. 1 root root  61M Jan 19 19:11 k3s
lrwxrwxrwx. 1 root root    3 Jan 19 19:11 kubectl -> k3s
lrwxrwxrwx. 1 root root    3 Jan 19 19:11 ctr -> k3s
-rwxr-xr-x. 1 root root 1.8K Jan 19 19:11 k3s-killall.sh
-rwxr-xr-x. 1 root root 1.4K Jan 19 19:11 k3s-uninstall.sh
```

**Reply by brandond** [reply]
You might need to run `hash -r` or `rehash` in your script after running the installer? This isn't generally necessary, but it's possible that your shell is just not refreshing the list of available commands.
